primitives/core/src/offchain/storage.rs

```diff
@@ -140,23 +141,21 @@ impl OffchainOverlayedChanges {
 	/// Remove a key and its associated value from the offchain database.
 	pub fn remove(&mut self, prefix: &[u8], key: &[u8]) {
 		if let Self::Enabled(ref mut storage) = self {
-			let key: Vec<u8> = prefix.iter().chain(key).cloned().collect();
-			let _ = storage.insert(key, OffchainOverlayedChange::Remove);
+			let _ = storage.insert((prefix.to_vec(), key.to_vec()), OffchainOverlayedChange::Remove);
 		}
 	}
 
 	/// Set the value associated with a key under a prefix to the value provided.
 	pub fn set(&mut self, prefix: &[u8], key: &[u8], value: &[u8]) {
 		if let Self::Enabled(ref mut storage) = self {
-			let key = prefix.iter().chain(key).cloned().collect();
-			let _ = storage.insert(key, OffchainOverlayedChange::SetValue(value.to_vec()));
+			let _ = storage.insert((prefix.to_vec(), key.to_vec()), OffchainOverlayedChange::SetValue(value.to_vec()));
 		}
 	}
 
 	/// Obtain a associated value to the given key in storage with prefix.
 	pub fn get(&self, prefix: &[u8], key: &[u8]) -> Option<OffchainOverlayedChange> {
 		if let Self::Enabled(ref storage) = self {
-			let key: Vec<u8> = prefix.iter().chain(key).cloned().collect();
+			let key = (prefix.to_vec(), key.to_vec());
```

Comment:
@tomusdrw the initial motivation was to avoid having to revert the prefix/key glue here, but also the fact that using a `/` might be a bad idea and allow aliasing to easily - using the tuple is guaranteed to be unique.